### DISCUSSION TITLE
Credentials CSFR Token not generating

### DISCUSSION THREAD

**Discussion by connermurphy** [nextauthjs/next-auth#4857] [Help]
### Question 💬

Credentials sign-in form (password only) not generating CSFR Token when deployed to vercel.

My project works fine locally, but when deployed I can't login. After inspecting, I can see hidden csfr input on form has empty value.

What could be the cause?

`NEXTAUTH_SECRET` is set.

### How to reproduce ☕️

```jsx
import LoginPage from "../../components/User/LoginPage";

import Head from "next/head";

import { getCsrfToken } from "next-auth/react";

import { getSession } from "next-auth/react";

function Index({ csrfToken }) {
  return (
    <>
      <Head>
        <title>Login</title>
      </Head>
      <LoginPage token={csrfToken} />
    </>
  );
}

export async function getServerSideProps(context) {
  const session = await getSession(context);

  if (!session) {
    return {
      props: {
        csrfToken: await getCsrfToken(context),
      },
    };
  } else {
    return {
      redirect: {
        permanent: false,
        destination: `/`,
      },
      props: {},
    };
  }
}

export default Index;

```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by sync-adm** [comment]
+1 for me it doesn't work in production on VERCEL either. Have you found any solution for this @connermurphy?
